opm/simulators/wells/GasLiftStage2.cpp

```diff
@@ -517,10 +543,9 @@ void
 GasLiftStage2::
 optimizeGroup_(const Group &group)
 {
-    const auto &gl_group = this->glo_.group(group.name());
-    const auto &max_glift = gl_group.max_lift_gas();
-    const auto &max_total_gas = gl_group.max_total_gas();
-    if (group.has_control(Group::ProductionCMode::ORAT)
+    const auto max_glift = getGroupMaxALQ_(group);
+    const auto max_total_gas = getGroupMaxTotalGas_(group);
+    if (group.has_control(Group::ProductionCMode::ORAT) || group.has_control(Group::ProductionCMode::LRAT)
                        || max_glift || max_total_gas)
```

Comment:
I think it would be better to check if the groups actual control is ORAT or LRAT not just that it has a control. 
i.e. `group_state.production_control(group.name()) == Group::ProductionCMode::ORAT`
I guess you then need to store const GroupState &group_state in GasLiftCommon.hpp

Reply:
Shouldn' t the logic in for example `removeSurplusALQ_()` also be done if the group  just has the control? That is, it does not need to be the actual control for the group. As an example, if the actual control of the group is WRAT but it also has a LRAT control, we should still remove surplus alq, see:

https://github.com/OPM/opm-simulators/blob/d31048d2987f6985ac1fef63ccd5527cfffa26aa/opm/simulators/wells/GasLiftStage2.cpp#L766

Reply:
My understanding is that it only make sense to try to optimize the distribution of the gaslift if the amount of gaslift is constraint either by the maximum allowed gaslift or total gas or the group is under individual control i.e. so maybe a better logic will be to enter stage 2 if `group_state.production_control(group.name()) != Group::ProductionCMode::NONE && group_state.production_control(group.name()) != FLD`

Reply:
> I have updated the PR to include this logic

The latest update leaves [`max_glift`](https://github.com/OPM/opm-simulators/blob/234750a563e3e353bf1ffd388f82e020841e84b8/opm/simulators/wells/GasLiftStage2.cpp#L547) and [`max_total_gas`](https://github.com/OPM/opm-simulators/blob/234750a563e3e353bf1ffd388f82e020841e84b8/opm/simulators/wells/GasLiftStage2.cpp#L548) [unused](https://ci.opm-project.org/job/opm-simulators-PR-builder/4159/gcc/new/).  Is that intentional because they will be used in a future update?  If not, you should probably remove the variables.

Reply:
> you should probably remove the variables

@bska Yes, I have updated the PR.